azure-client-runtime/src/main/java/com/microsoft/azure/PollingState.java

```diff
@@ -141,28 +202,50 @@ void updateFromResponseOnDeletePost(Response<ResponseBody> response) throws IOEx
      */
     int delayInMilliseconds() {
         if (this.retryTimeout >= 0) {
-            return this.retryTimeout * 1000;
+            return this.retryTimeout;
         }
-        if (this.response != null && response.headers().get("Retry-After") != null) {
-            return Integer.parseInt(response.headers().get("Retry-After")) * 1000;
+        if (this.defaultRetryTimeout >= 0) {
+            return this.defaultRetryTimeout * 1000;
         }
         return AzureAsyncOperation.DEFAULT_DELAY * 1000;
     }
 
     /**
-     * Gets the polling status.
-     *
-     * @return the polling status.
+     * @return the uri of the resource on which the LRO PUT or PATCH applied.
      */
-    String status() {
-        return status;
+    String putOrPatchResourceUri() {
+        return this.putOrPatchResourceUri;
     }
 
     /**
-     * @return the resource type
+     * @return true if the status this state hold represents terminal status.
      */
-    Type resourceType() {
-        return resourceType;
+    boolean isStatusTerminal() {
```

Comment:
yes, thats what i thought while implementing theses methods [isStatusTerminal, isStatusFailed, isStatusSucceeded]. We expose 'status' to user. Do you think it is useful for user to expose these methods?